engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -121,11 +135,18 @@ func (pool *EnclavePool) GetEnclave(
 		return nil, nil
 	}
 
+	// If there is no idle enclave in the pool returns nil
+	// for not to block the caller
```

Comment:
Exactly, here you have some overhead because you're using a channel, I think a list would be enough here

Reply:
I put my comment about this above, we can discuss it